gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/step/map/OrderGlobalStep.java

```diff
@@ -99,6 +99,19 @@ public void modulateBy(final Traversal.Admin<?, ?> traversal, final Comparator c
     }
 
     @Override
+    public void replaceLocalChild(final Traversal.Admin<?, ?> oldTraversal, final Traversal.Admin<?, ?> newTraversal) {
+        int i = 0;
+        for (final Pair<Traversal.Admin<S, C>, Comparator<C>> pair : this.comparators) {
+            final Traversal.Admin<S, C> traversal = pair.getValue0();
+            if (null != traversal && traversal.equals(oldTraversal)) {
+                this.comparators.set(i, Pair.with(this.integrateChild(newTraversal), pair.getValue1()));
```

Comment:
I would add a test case to `OrderTest` (if it doesn't already exist) that verifies that your strategy works as `OrderXXXStep` is complicated with how it deals with comparators. Thus, add a `...order().by(label())...`-style test so we know all is good (in the proverbial hood).